packages/datetime/test/dateRangeInputTests.tsx

```diff
@@ -487,6 +488,1304 @@ describe("<DateRangeInput>", () => {
+                        assertInputTextsEqual(root, HOVER_TEST_DATE_CONFIG.str, "");
+                    });
+
+                    it("keeps focus on start field", () => {
+                        assertStartInputFocused(root);
+                    });
+
+                    describe("on click", () => {
+
+                        beforeEach(() => {
+                            dayElement.simulate("click");
+                        });
+
+                        it("sets selection to [<hoveredDate>, null]", () => {
+                            assertInputTextsEqual(root, HOVER_TEST_DATE_CONFIG.str, "");
+                        });
+
+                        it("moves focus to end field", () => {
+                            assertEndInputFocused(root);
+                        });
+                    });
+
+                    describe("if mouse moves to no longer be over a calendar day...", () => {
+
+                        beforeEach(() => {
+                            dayElement.simulate("mouseleave");
+                        });
+
+                        it("shows [null, null] in input fields", () => {
+                            assertInputTextsEqual(root, "", "");
+                        });
+
+                        it("keeps focus on start field", () => {
+                            assertStartInputFocused(root);
+                        });
+                    });
+                });
+
+                describe("if end field is focused...", () => {
+
+                    beforeEach(() => {
+                        endInput.simulate("focus");
+                        dayElement.simulate("mouseenter");
+                    });
+
+                    it("shows [null, <hoveredDate>] in input fields", () => {
+                        assertInputTextsEqual(root, "", HOVER_TEST_DATE_CONFIG.str);
+                    });
+
+                    it("keeps focus on end field", () => {
+                        assertEndInputFocused(root);
+                    });
+
+                    it("sets selection to [null, <hoveredDate>] on click", () => {
+                        dayElement.simulate("click");
+                        assertInputTextsEqual(root, "", HOVER_TEST_DATE_CONFIG.str);
+                    });
+
+                    describe("on click", () => {
+
+                        beforeEach(() => {
+                            dayElement.simulate("click");
+                        });
+
+                        it("sets selection to [null, <hoveredDate>]", () => {
+                            assertInputTextsEqual(root, "", HOVER_TEST_DATE_CONFIG.str);
+                        });
+
+                        it("moves focus to start field", () => {
+                            assertStartInputFocused(root);
+                        });
+                    });
+
+                    describe("if mouse moves to no longer be over a calendar day...", () => {
+
```

Comment:
i'm personally not a fan of this leading whitespace in functions. not having it makes nested blocks a little easier to visually parse.